The eigen_spatial_convolutions_test benchmark crashes

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): b'v1.8.0-14450-g9c38906' 1.13.1
- Python version: Python 3.5.2
- Bazel version (if compiling from source): 0.19.2
- GCC/Compiler version (if compiling from source): g++ (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**

```
$ bazel test --config opt -- //tensorflow/core/kernels:eigen_spatial_convolutions_test

Starting local Bazel server and connecting to it...
DEBUG: /home/user/.cache/bazel/_bazel_user/59a925eeb655b30b5c683f8317fe569e/external/build_bazel_rules_apple/apple/repositories.bzl:35:5: 
WARNING: `build_bazel_rules_apple` depends on `bazel_skylib` loaded from https://github.com/bazelbuild/bazel-skylib.git (tag 0.6.0), but we have detected it already loaded into your workspace from None (tag None). You may run into compatibility issues. To silence this warning, pass `ignore_version_differences = True` to `apple_rules_dependencies()`.

INFO: Analysed target //tensorflow/core/kernels:eigen_spatial_convolutions_test (78 packages loaded, 3781 targets configured).
INFO: Found 1 test target...
INFO: From Compiling tensorflow/core/kernels/eigen_spatial_convolutions_test.cc:
tensorflow/core/kernels/eigen_spatial_convolutions_test.cc: In function 'void Eigen::PackRhsHelper(int, int, int, int, int, int, int, int, int, int, Eigen::Index, Eigen::Index)':
tensorflow/core/kernels/eigen_spatial_convolutions_test.cc:1400:69: warning: typedef 'using Traits = struct Eigen::internal::gebp_traits<float, float>' locally defined but not used [-Wunused-local-typedefs]
   using Traits = typename Eigen::internal::gebp_traits<float, float>;
                                                                     ^
tensorflow/core/kernels/eigen_spatial_convolutions_test.cc: In function 'void Eigen::PackLhsHelper(int, int, int, int, int, Eigen::Index, Eigen::Index)':
tensorflow/core/kernels/eigen_spatial_convolutions_test.cc:1631:16: warning: variable 'nocontract_dim' set but not used [-Wunused-but-set-variable]
   nocontract_t nocontract_dim = {0};
                ^
tensorflow/core/kernels/eigen_spatial_convolutions_test.cc:1632:14: warning: variable 'contract_dim' set but not used [-Wunused-but-set-variable]
   contract_t contract_dim = {1};
              ^
Target //tensorflow/core/kernels:eigen_spatial_convolutions_test up-to-date:
  bazel-bin/tensorflow/core/kernels/eigen_spatial_convolutions_test
INFO: Elapsed time: 36.190s, Critical Path: 31.21s, Remote (0.00% of the time): [queue: 0.00%, setup: 0.00%, process: 0.00%]
INFO: 25 processes: 25 local.
INFO: Build completed successfully, 27 total actions
//tensorflow/core/kernels:eigen_spatial_convolutions_test                PASSED in 0.2s

INFO: Build completed successfully, 27 total actions

$ bazel-bin/tensorflow/core/kernels/eigen_spatial_convolutions_test --benchmarks=all
Running main() from test_main.cc
Benchmark                                                        Time(ns) Iterations
------------------------------------------------------------------------------------
*** Received signal 11 ***
*** BEGIN MANGLED STACK TRACE ***
/home/user/.cache/bazel/_bazel_user/59a925eeb655b30b5c683f8317fe569e/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/core/kernels/../../../_solib_k8/_U_S_Stensorflow_Score_Skernels_Ceigen_Uspatial_Uconvolutions_Utest___Utensorflow/libtensorflow_framework.so(+0x83085e)[0x7f470a1d785e]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x11390)[0x7f4708fdf390]
bazel-bin/tensorflow/core/kernels/eigen_spatial_convolutions_test[0x40c6cc]
bazel-bin/tensorflow/core/kernels/eigen_spatial_convolutions_test[0x412c58]
bazel-bin/tensorflow/core/kernels/eigen_spatial_convolutions_test[0x41316f]
/home/user/.cache/bazel/_bazel_user/59a925eeb655b30b5c683f8317fe569e/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/core/kernels/../../../_solib_k8/libtensorflow_Score_Slibtest.so(_ZN10tensorflow7testing9Benchmark3RunEiiPiPd+0x44)[0x7f470ab601e4]
/home/user/.cache/bazel/_bazel_user/59a925eeb655b30b5c683f8317fe569e/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/core/kernels/../../../_solib_k8/libtensorflow_Score_Slibtest.so(_ZN10tensorflow7testing9Benchmark3RunEPKc+0x409)[0x7f470ab612d9]
/home/user/.cache/bazel/_bazel_user/59a925eeb655b30b5c683f8317fe569e/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/core/kernels/../../../_solib_k8/libtensorflow_Score_Slibtest_Umain.so(main+0xb3)[0x7f470ab65b83]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7f4708395830]
bazel-bin/tensorflow/core/kernels/eigen_spatial_convolutions_test[0x409649]
*** END MANGLED STACK TRACE ***

*** Begin stack trace ***
	tensorflow::CurrentStackTrace[abi:cxx11]()
	
	
	
	
	
	tensorflow::testing::Benchmark::Run(int, int, int*, double*)
	tensorflow::testing::Benchmark::Run(char const*)
	main
	__libc_start_main
	
*** End stack trace ***
Aborted (core dumped)
```

**Describe the expected behavior**

The benchmark should run correctly and should not crash.

**Code to reproduce the issue**

```
$ bazel test --config opt -- //tensorflow/core/kernels:eigen_spatial_convolutions_test
$ bazel-bin/tensorflow/core/kernels/eigen_spatial_convolutions_test --benchmarks=all
```

The issue is 100% reproducible on my machine

--- a/tensorflow/core/kernels/eigen_spatial_convolutions_test.cc
+++ b/tensorflow/core/kernels/eigen_spatial_convolutions_test.cc
@@ -1468,6 +1468,10 @@ static void PackRhsHelper(int iters,
   std::vector<Evaluator> evaluators;
   std::vector<InputMapper> input_mappers;
 
+  inputs.reserve(num_inputs);
+  evaluators.reserve(num_inputs);
+  input_mappers.reserve(num_inputs);
+
   for (int i = 0; i < num_inputs; ++i) {
     inputs.emplace_back(input_dims);
     inputs[i].setRandom();
@@ -1651,6 +1655,10 @@ static void PackLhsHelper(int iters,
   std::vector<Tensor<float, 4>> filters;
   std::vector<Evaluator> evaluators;
   std::vector<InputMapper> input_mappers;
+
+  filters.reserve(num_filters);
+  evaluators.reserve(num_filters);
+  input_mappers.reserve(num_filters);
 
   for (int i = 0; i < num_filters; ++i) {
     filters.emplace_back(filter_dims);

